### DISCUSSION TITLE
Why are Midi functions 4 bytes

### DISCUSSION THREAD

**Discussion by wapordev** [hathach/tinyusb#1908] [Q&A]
My program did not work, and eventually I figured out I had to access the midi status from packet[1] instead of packet[0]. Why do the functions send and receive 4 bytes? Aren't midi messages 3, with 1 status and 2 data bytes? What is in this first byte that I apparently just have to skip?

To my knowledge there is no documentation for these functions, nor does the code for them even have comments. If this is available somewhere feel free to send a link.

**Comment by hathach** [comment]
midi packet is 4 bytes, it is part of USB MIDI specs. You could use the old midi_stream() API which is designed to simulate original (serial) 3 bytes midi.

**Comment by wapordev** [comment]
Ok thank you I will look more into it then. I guess the results I found were outdated then or something

**Comment by rppicomidi** [comment]
@wapordev For all the details of the USB MIDI 1.0 spec, see the [official class spec](https://www.usb.org/sites/default/files/midi10.pdf). If you want the full MIDI 1.0 spec, you have to join MIDI.org (it's free) and then download it [here](https://midi.org/specifications/midi1-specifications). Is this what you were asking?

**Comment by wapordev** [comment]
Yes that's what I was asking for. My main issue has been solved, as in getting MIDI working at all, but once I get my main features working, and get a controller that I can actually test the new ones with, I will probably add per note pitch shifting. A bit annoying that you have to sign up for a website to access a standard, but oh well.
